packages/account/src/BiconomySmartAccount.ts

```diff
@@ -72,9 +74,8 @@ export class BiconomySmartAccount extends SmartAccount implements IBiconomySmart
     try {
       this.isProviderDefined()
       this.isSignerDefined()
-      this.setAccountIndex(accountIndex)
+      await this.setAccountIndex(accountIndex)
       this.chainId = await this.provider.getNetwork().then((net) => net.chainId)
-      await this.getInitCode(accountIndex)
```

Comment:
why is this commented?

Reply:
its not commented as its just moved from here to getSmartAccountAddress() function so that where ever setAccountIndex is called on it will fetch again initCode for new smart account